### ISSUE TITLE
CLIP-ReID preview HERE!

### ISSUE THREAD

**Issue by mikel-brostrom** [mikel-brostrom/yolo_tracking#1029]
### Search before asking

- [X] I have searched the Yolov8 Tracking [issues](https://github.com/mikel-brostrom/yolo_tracking/issues) and found no similar enhancement requests.

### Description

Some tracking preview using clip-reid!

```
git clone https://github.com/mikel-brostrom/yolo_tracking
git checkout clip-reid
python examples/track.py --reid-model clip_market1501.pt
# or
python examples/track.py --reid-model clip_duke.pt
```

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by mikel-brostrom** [comment]
Not yet. Don't know when I will have time for this...

**Comment by mj2323** [comment]
Thank you for your hard work. Mikel.

Added a pretty interesting SOTA algorithm.
I'm not sure how much the text encoder will improve compared to existing Re-ID performance, but I'm excited because this work could be experimentally meaningful.

**Comment by mikel-brostrom** [comment]
Yes! I wanted to add a ReID model trained on (image, text) pairs just for the sake of comparison. I had sadly not had the time to experiment with it so if you have some results I would be glad to discuss them :smile:

**Comment by mj2323** [comment]
There are a total of 4 Re-ID models(OS-Net, ResNet-50, MLFN, CLIP-ReID) that I used, and I thought once again that your repository is valuable from a research perspective. In conclusion, compared to existing image encoder-based Re-ID models, CLIP has a very precise similarity value to distinguish between other person.

CLIP's own embedding vector value seems more unique than MOT performance indicators such as IDF1.

**Comment by mikel-brostrom** [comment]
So you see IDF1 boosts ising CLIP compared to the rest of the ReID models?

**Comment by mj2323** [comment]
> So you see IDF1 boosts ising CLIP compared to the rest of the ReID models @mj2323 ?

Sorry for the late reply.

I am currently working on object tracking with multiple input videos through batch-processing.

Therefore, it is necessary to determine whether the object for multi-input is the same person or a different person. In conclusion, CLIP's performance is excellent.

When two inputs are batch processed as shown below, a tracking ID for each batch appears. When two inputs are batch processed as shown below, a tracking ID for each batch appears. When you actually check the video, ID:2 in batch_0_id_list and ID:1 in batch_1_id_list are the same person. Likewise, ID:4 in batch_0_id_list and ID:2 in batch_1_id_list are the same person.

```
batch_0_id_list = [2, 4]

batch_1_id_list = [1, 2]
```
The table below shows the similarity of the same person for each batch video. The closer the similarity value is to 0, the higher the probability that it is the same person.
![image](https://github.com/mikel-brostrom/yolo_tracking/assets/67301414/2c805d87-1bf9-4f06-b285-a3e9b8be660b)


The table below shows the similarity to other people for each batch video. The closer the similarity value is to 1, the higher the probability that it is a different person.
![image](https://github.com/mikel-brostrom/yolo_tracking/assets/67301414/575fffbb-ef38-4b25-a238-252c3025b974)

In conclusion, I don't think the CLIP algorithm improves the IDF1 numbers much. However, CLIP's method of expressing embedding vectors has been confirmed to produce quite unique results.

**Comment by mikel-brostrom** [comment]
Great experiment @mj2323! Thank you for sharing! Yes, it is clear from you results that Clip-REID separates instances in embedding space much better for the task of ReID and this use-case than OSnet, ResNet50 and MLFN.
